ui/lib/utils/i18n.ts

```diff
@@ -8,7 +8,11 @@ import { initReactI18next } from 'react-i18next'
 
 i18next.on('languageChanged', function (lng) {
   console.log('Language', lng)
-  dayjs.locale(lng.toLowerCase())
+  if (lng.startsWith('zh')) {
```

Comment:
Is it better to revert some the changes in #677? IMO it is better to leave some capability to support both zhCN and zhTW. Will it work if we fallback `zh` to `zh-CN`?

https://www.i18next.com/principles/fallback#fallback-language

Reply:
Now we support zhTW and others zhXxx languages, if you add `zh-TW.yml`, then it will use the zh-TW translations first if the language is zh-TW, and fallback to zh and en when the translation doesn't exist in the zh-TW.yml.

Tried the fallback-language setting, it's a bit troublesome, you need to cover many languages, for example:

```
fallbackLng: { 
    'zh-HK': ['zh-CN', 'en'], 
    'zh-TW': ['zh-CN', 'en'],
    'zh-SG': ...,
    'zh-Hans': ...,
    'zh-Hant': ...,
    ...
}
```

According to some documents, the language may be: 

```
zh-Hans 简体中文
zh-Hans-CN 大陆地区使用的简体中文
zh-Hans-HK 香港地区使用的简体中文
zh-Hans-MO 澳门使用的简体中文
zh-Hans-SG 新加坡使用的简体中文
zh-Hans-TW 台湾使用的简体中文
zh-Hant 繁体中文
zh-Hant-CN 大陆地区使用的繁体中文
zh-Hant-HK 香港地区使用的繁体中文
zh-Hant-MO 澳门使用的繁体中文
zh-Hant-SG 新加坡使用的繁体中文
zh-Hant-TW 台湾使用的繁体中文
```

(Although I don't meet)

Reply:
I think maybe we can fix this bug first. Then later if someone contributes the traditional Chinese translation for us, we can add the support for dayjs.

Reply:
This fallback looks to be stupid. Why can't we just fallback zh to zh-CN?

Reply:
Not sure whether `zh -> zh-CN` will make `zh-TW` fallback to `zh-CN`, let me have a try.

Reply:
Tried to revert it and use the following setting for i18next: 

```
export const ALL_LANGUAGES = {
  'zh-CN': '简体中文',
  en: 'English',
}

i18next
  .use(LanguageDetector)
  .use(initReactI18next)
  .init({
    resources: {}, // oh! this line is a big pitfall, we can't remove it, else it will cause strange crash!
    fallbackLng: {
      zh: ['zh-CN'],
      default: ['en'],
    },
    interpolation: {
      escapeValue: false,
    },
  })

```

It works in most cases, except when you run the dashboard for the first time with another language likes `zh-TW`.

![企业微信20200917112000](https://user-images.githubusercontent.com/1284531/93416704-e366a200-f8d8-11ea-9a5a-8028a6150d85.png)

![image](https://user-images.githubusercontent.com/1284531/93416761-01340700-f8d9-11ea-9a76-f68408044b6a.png)

Because we don't import zh-tw locale for dayjs, and i18next's fallback doesn't affect dayjs.

But it will always works fine after manually selecting the 'zh-CN' or 'en' language.

Reply:
I also tried this:

```
import en from 'dayjs/locale/en'
import zh from 'dayjs/locale/zh-cn'

//...

const DAYJS_LOCALES = { zh, en }
// console.log(DAYJS_LOCALES)
i18next.on('languageChanged', function (lng) {
  dayjs.locale(DAYJS_LOCALES[lng.toLocaleLowerCase()] || en)
})
```

It works when language is `zh`, but crashes when `en`, I think the dayjs has bug (will check it later).

Reply:
Seems to find a good way:

```
import zh from 'dayjs/locale/zh-cn'

const DAYJS_LOCALES = { zh }

i18next.on('languageChanged', function (lng) {
  dayjs.locale(DAYJS_LOCALES[lng.toLocaleLowerCase()] || 'en')
})
```

Reply:
Maybe the core problem is that, the initial language is the *detected* language, instead of the *effective* language.

Reply:
Sorry, I am a little confused 😂. So you mean we need write a util method `getEffectiveLang()` and use its value as Select component's default value, right?

I tried it and works fine, and we don't need to revert the zh.yml to zh-CN.yml.

```
export function getEffetiveLang(): string {
  const effetiveLangs = Object.keys(ALL_LANGUAGES)
  const detectedLang = i18next.language
  if (effetiveLangs.includes(detectedLang)) {
    return detectedLang
  }
  if (detectedLang.startsWith('zh')) {
    return 'zh-CN'
  }
  return 'en'
}

        <Form layout="vertical" initialValues={{ language: getEffetiveLang() }}>
          <Form.Item name="language" label={t('user_profile.i18n.language')}>
            <Select onChange={handleLanguageChange} style={{ width: 200 }}>
              {_.map(ALL_LANGUAGES, (name, key) => {
                return (
                  <Select.Option key={key} value={key}>
                    {name}
                  </Select.Option>
                )
              })}
            </Select>
          </Form.Item>
        </Form>
```